spec/classes/profile/keepalived_spec.rb

```diff
@@ -0,0 +1,96 @@
+      let(:base_file) { '/etc/keepalived/keepalived.conf' }
+      let(:service) { 'keepalived' }
+
+      include_context 'with mocked puppetdb functions', 'somedc', %w[thisnode scotch soda]
+
+      describe 'packages' do
+        it { is_expected.to contain_package('keepalived') }
+        it { is_expected.to contain_package('ipset') }
+      end
+
+      describe 'sysctl conf' do
+        let(:file) { '/etc/sysctl.d/keepalived.conf' }
+
+        it { is_expected.to contain_file(file).with(ensure: 'present') }
+        it { is_expected.to contain_file(file).with(mode: '0644') }
+
+        it 'says it is managed by puppet' do
+          is_expected.to contain_file(file).with_content(
+            %r{\A# Managed by puppet \(nebula\/profile\/keepalived\/sysctl\.conf\.erb\)\n},
+          )
+        end
+
+        it 'enables ip_nonlocal_bind' do
+          is_expected.to contain_file(file).with_content(%r{^net.ipv4.ip_nonlocal_bind = 1$})
+        end
+      end
+
+      describe 'base config file' do
+        let(:file) { base_file }
+
+        it { is_expected.to contain_file(file).with(ensure: 'present') }
+        it { is_expected.to contain_file(file).with(require: 'Package[keepalived]') }
+        it { is_expected.to contain_file(file).with(notify: 'Service[keepalived]') }
+        it { is_expected.to contain_file(file).with(mode: '0644') }
+
+        it 'has a vrrp_scripts check_haproxy section' do
+          is_expected.to contain_file(file).with_content(%r{^vrrp_script check_haproxy})
+        end
+
+        it 'has the haproxy floating ip address' do
+          is_expected.to contain_file(file).with_content(%r{virtual_ipaddress {\n\s*12\.23\.32\.22\n\s*}}m)
+        end
+
+        context 'with a floating ip address parameter' do
+          let(:params) { { floating_ip: Faker::Internet.ip_v4_address } }
+
+          it { is_expected.to contain_file(file).with_content(%r{virtual_ipaddress {\n\s*#{params[:floating_ip]}\n\s*}}m) }
+        end
+
+        it { is_expected.to contain_file(file).with_content(%r{unicast_src_ip #{my_ip}}) }
+
+        it 'has a unicast_peer block with the IP addresses of all nodes with the same profile at the same datancenter except for me' do
+          is_expected.to contain_file(file).with_content(%r{unicast_peer {\n\s*#{scotch['ip']}\n\s*#{soda['ip']}\n\s*}})
+        end
+
+        it { is_expected.to contain_file(file).with_content(%r{interface #{facts[:networking][:primary]}}) }
+
+        it { is_expected.to contain_file(file).with_content(%r{notification_email {\n\s.*root@default.invalid\n\s.*}}m) }
+        it { is_expected.to contain_file(file).with_content(%r{notification_email_from root@default.invalid}) }
+      end
+
+      describe 'service' do
+        it { is_expected.to contain_service(service).that_requires('Package[keepalived]') }
+        xit { is_expected.to contain_service(service).that_requires('Package[haproxyctl]') }
```

Comment:
Why the skipped test?

Reply:
Relates to the commented out dependency in keepalived.pp earlier - we had wanted to remove the dependency on the haproxy profile but I don't remember what the specific issue was. I'll remove this test for now. We may or may not want to put this dependency back in some form later.

Reply:
The worst case if keepalived gets installed without haproxyctl is that all nodes will fail their check, which is probably fine, b/c if haproxyctl isn't installed that probably means haproxy isn't either...

Reply:
This kind of reflects that the profile is misnamed - it isn't really keepalived, it's keepalived specifically for haproxy.